### ISSUE TITLE
Sluttid on standardrutter and maybe other schemes 

### ISSUE THREAD

**Issue by ekol-ali** [biodiversitydata-se/biocollect#239]
In BioCollect there must be a starttid and sluttid on standardrutter. When starting in 1996, surveyors were asked to give this information on the paper protocols then used.  HOWEVER, **it was never entered into the database** (which later became totalstandard). On the excel-files that we started to use later (mid 2000s), and still do, starttid and sluttid were/are not asked for. Thus, until we started using BioCollect, starttid and sluttid never entered the database. But now it is required in BioCollect.

Annelie told me that there is an algorithm that creates sluttid by using the latest time for a point and then adding 5 minutes. This is of course better than nothing, but will rarely be correct. Most routes end after a full line is carried out. This is about 30-40 min after the last point was made. Thus, a better algorithm would be: take last point carried out and **add 35 minutes** (30 for the line and 5 for the point.

This wil still not be correct in many cases, since some routes actually earlier than this, everything from ending directly after a point (when 5 minutes addition would be correct) to ending with a 900 m line (when 30 minutes addition would be correct). Due to how the  routes are configurated, or the decision of the surveyor, it will NEVER be possible to have a perfect algorithm for this. 

One alternative is of course to go back to all 5000 (?) paper copies and add this manually.... [No, we shouldn't]

**Comment by mathieuLU** [comment]
ok several comments here :

 - if needed i think we can remove the fields starttime/endtime (to be confirmed), or at least make them optional.
 - this +5 minutes is done when importing from excel. No problem to put +35 instead. ok ?

if needed i can change the +5 in +35 for all the historical records.

**Comment by ekol-ali** [comment]
I think it is fine to **keep** starttime/endtime, it makes a lot of sense to have that info.
The problem is the older ones, that we can in principle never get correct in a systematic way. 
So in the Excel import you look for the latest point-time and add 5 min? 
Yes - please make it +35 minutes instead and change the historical ones to +35 as well.

**Comment by mathieuLU** [comment]
Now the intranet script adds 35 minutes to the finish time, instead of 5 minutes. For standardrutter only.
https://github.com/Lund-University-Biodiversity-data/intranet-ekol/commit/de6cfe2c77a8185b5d86bbb57b8bc87935f97070
For nattrutterna, we stick to 5 minutes, right ?

Now I have to fix the historical records, added from that script. Since you don't use this data (surveyFinishTIme), i guess you never change this field afterwards ?

so for all the STD activities registered with "dataOrigin" : "scriptExcel", i will add 30min to the exsting finishTime.
Right ?

**Comment by ekol-ali** [comment]
surveyFinishTime is admittedly something we do not make much use of. But again, I really think **it makes sense** to have the observers to report it.
For nattrutter, +5 from the starttime of the point surveyed at the latest time, is fine.
Your last question: I guess you mean **35** min? 35 minutes should be added to the the starttime of the point surveyed at the latest time. This will, on average, come closest, because most routes end with a 1 km transect, that suppossedly will take 30 minutes. So, 5 min for the last point,and then 30 min for the last km.

**Comment by mathieuLU** [comment]
yes. What i meant for the historical records is that we have already added 5 minutes to the finish time when we imported them from the excel script. To make it relevant, we have to add now 30min, to get +35 in total

ANd we have as well the historical records coming from the PSQL database. For them as well, just like the excel import script, we had added +5 minutes. So again we have to change this to +35, by adding 30 minutes to the current finishTime

right ?
